### PR TITLE
[T3A4][T14-A1]Yeo Kai Yan

### PR DESCRIPTION
**Pull request by kaiyanyeo**

### CODE REVIEW

**Comment by kaiyanyeo** [comment]
Ready for review

File: src/seedu/addressbook/Main.java [hunk 1]
```diff
@@ -31,12 +33,12 @@
     private List<? extends ReadOnlyPerson> lastShownList = Collections.emptyList();
 
 
-    public static void main(String... launchArgs) {
+    public static void main(String... launchArgs) throws StorageFileDeletedException {
```
Comment (okkhoy) [inline src/seedu/addressbook/Main.java]: this should not be the case. if main throws, then the program halts, right? who will catch this exception?

please recollect the discussion around this during the tutorial and try to make amends going further. no need to resubmit this PR this time.

File: src/seedu/addressbook/Main.java [hunk 2]
```diff
@@ -103,16 +105,19 @@ private void recordResult(CommandResult result) {
      * @param command user command
      * @return result of the command
      */
-    private CommandResult executeCommand(Command command)  {
+    private CommandResult executeCommand(Command command) throws StorageFileDeletedException {
         try {
             command.setData(addressBook, lastShownList);
             CommandResult result = command.execute();
-            storage.save(addressBook);
+            storage.save(addressBook, true);
```
Comment (okkhoy) [inline src/seedu/addressbook/Main.java]: this one is a little confusing. why do you want to modify the default way of handling the save method?

File: src/seedu/addressbook/Main.java [hunk 3]
```diff
@@ -103,16 +105,19 @@ private void recordResult(CommandResult result) {
...
             return result;
+        } catch (StorageFileDeletedException e){
```
Comment (okkhoy) [inline src/seedu/addressbook/Main.java]: this is acceptable. but i don't understand if you are doing this, why are you still throwing exceptions to higher classes?

File: src/seedu/addressbook/storage/StorageFile.java [hunk 4]
```diff
@@ -83,13 +85,18 @@ private static boolean isValidPath(Path filePath) {
...
      * @throws StorageOperationException if there were errors converting and/or storing data to file.
      */
-    public void save(AddressBook addressBook) throws StorageOperationException {
+    public void save(AddressBook addressBook, boolean wasRunning) throws StorageOperationException, StorageFileDeletedException {
 
         /* Note: Note the 'try with resource' statement below.
          * More info: https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html
          */
+        File storageFile = path.toFile();
+        if (wasRunning && !storageFile.exists()){
```
Comment (okkhoy) [inline src/seedu/addressbook/storage/StorageFile.java]: - the check is correct
- this implementation stops the program flow and halts the program
 - better way is to let the program create the file as it was doing earlier, but use exception as a mechanism to let the user know of the fact that the file was deleted.

File: src/seedu/addressbook/storage/StorageFile.java [hunk 5]
```diff
@@ -107,8 +114,9 @@ public void save(AddressBook addressBook) throws StorageOperationException {
      * Loads data from this storage file.
      *
      * @throws StorageOperationException if there were errors reading and/or converting data from file.
+     * @throws StorageFileDeletedException if the storage file is deleted when the program is running.
      */
-    public AddressBook load() throws StorageOperationException {
+    public AddressBook load() throws StorageOperationException, StorageFileDeletedException {
```
Comment (okkhoy) [inline src/seedu/addressbook/storage/StorageFile.java]: probably don't need this. 
load doesn't need to check for this exception.

try to understand the context of the problem, this exception is intended to handle accidental file deletes during runtime.

**Comment by okkhoy** [comment]
@kaiyanyeo this is a good attempt, i have provided some comments in the code on where the solution can be improved. Please take a look, ack and close the PR once you have read the comments.

**Pull request closed without merging**